### PR TITLE
Bearing data model endpoints

### LINKED ISSUE #31: Init data model of ball bearing
Init a data model for ball bearings:

- try to find a place for all parameters of a ball bearing

After creating the data model replace your fruits endpoint with the ball bearings data model to:
- get
- post
- update
- delete
bearings.

Do not add a external database. this will be done in a later step of this project.

### PR DESCRIPTION
**Pull request by TheMaxens**
I created the endpoints to
- GET (get all arrangements + get one arrangement by ID) 
- POST (create an arrangement)
- UPDATE (update an existing arrangement via ID) 
- DELETE (delete an existing arrangement via ID)

bearings/OArrangements. 

The ID is represented by the field `uid`.

It is best to test the endpoints in Swagger-UI (`/q/swagger-ui`).

fix #31

### CODE REVIEW

File: src/main/java/org/dhbw/mosbach/ai/Bearing.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    public Bearing(double cdyn, double y, double e, double xB1) {
+        this.cdyn = cdyn;
+        this.y = y;
+        this.e = e;
+        this.xB1 = xB1;
+    }
+
+    @JsonProperty
+    public double getCdyn() {
```
Comment (philippabele) [inline src/main/java/org/dhbw/mosbach/ai/Bearing.java:24]: There is a lot of boilerplate code in your classes. Please take a lot a this [book](https://github.com/philippabele/ball-bearings-with-quarkus/blob/8c2ea4f963d0833194a82485d4542643b74f89f8/docs/Bibliography.bib#L44) on page 132 (not PDF page) and extend your class with PanacheEntity.

May this [example](https://quarkus.io/guides/hibernate-orm-panache#first-an-example) helps you as well.

**Review by philippabele — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -2,4 +2,12 @@ quarkus.http.host=0.0.0.0
...
 quarkus.smallrye-openapi.info-contact-email=techsupport@example.com
 quarkus.smallrye-openapi.info-contact-name=Example API Support
-quarkus.smallrye-openapi.info-contact-url=http://exampleurl.com/contact
\ No newline at end of file
+quarkus.smallrye-openapi.info-contact-url=http://exampleurl.com/contact
+# configure your datasource
+quarkus.datasource.db-kind = postgresql
+quarkus.datasource.username = sarah
+quarkus.datasource.password = connor
+quarkus.datasource.jdbc.url = jdbc:postgresql://localhost:5432/mydatabase
```
Comment (philippabele) [inline src/main/resources/application.properties:10]: Do you have a local docker postgresql instance running?

By running your application with `quarkus dev` , I receive the following error

###############

root@4ffa65374055:/ball-bearings-with-quarkus# quarkus dev

BUILD SUCCESSFUL in 2s
7 actionable tasks: 1 executed, 6 up-to-date
Listening for transport dt_socket at address: 5005

2022-12-04 09:45:42,773 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure
2022-12-04 09:45:42,781 WARN  [io.qua.arc.impl] (Quarkus Main Thread) An error occurred during delivery of the @BeforeDestroyed(ApplicationScoped.class) event: javax.enterprise.inject.CreationException: Config root [io.quarkus.hibernate.orm.runtime.HibernateOrmRuntimeConfig] with config phase [RUN_TIME] not initialized yet.
        at io.quarkus.hibernate.orm.runtime.HibernateOrmRuntimeConfig_66229d6c2e25c4818b3ef00069aed3a5861e7464_Synthetic_Bean.create(Unknown Source)
        at io.quarkus.hibernate.orm.runtime.HibernateOrmRuntimeConfig_66229d6c2e25c4818b3ef00069aed3a5861e7464_Synthetic_Bean.get(Unknown Source)
        at io.quarkus.hibernate.orm.runtime.HibernateOrmRuntimeConfig_66229d6c2e25c4818b3ef00069aed3a5861e7464_Synthetic_Bean.get(Unknown Source)
        at io.quarkus.arc.impl.CurrentInjectionPointProvider.get(CurrentInjectionPointProvider.java:62)
        at io.quarkus.hibernate.orm.runtime.JPAConfig_Bean.create(Unknown Source)
        at io.quarkus.hibernate.orm.runtime.JPAConfig_Bean.create(Unknown Source)
        at io.quarkus.arc.impl.AbstractSharedContext.createInstanceHandle(AbstractSharedContext.java:113)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:37)
        at io.quarkus.arc.impl.AbstractSharedContext$1.get(AbstractSharedContext.java:34)
        at io.quarkus.arc.impl.LazyValue.get(LazyValue.java:26)
        at io.quarkus.arc.impl.ComputingCache.computeIfAbsent(ComputingCache.java:69)
        at io.quarkus.arc.impl.AbstractSharedContext.get(AbstractSharedContext.java:34)
        at io.quarkus.hibernate.orm.runtime.JPAConfig_Observer_destroy_b8200764d6ca5f0908a962ee9c3337f12042c096.notify(Unknown Source)
        at io.quarkus.arc.impl.EventImpl$Notifier.notifyObservers(EventImpl.java:326)
        at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:314)
        at io.quarkus.arc.impl.EventImpl$Notifier.notify(EventImpl.java:244)
        at io.quarkus.arc.impl.ArcContainerImpl.shutdown(ArcContainerImpl.java:397)
        at io.quarkus.arc.Arc.shutdown(Arc.java:58)
        at io.quarkus.arc.runtime.ArcRecorder$1.run(ArcRecorder.java:47)
        at io.quarkus.runtime.StartupContext.runAllInReverseOrder(StartupContext.java:84)
        at io.quarkus.runtime.StartupContext.close(StartupContext.java:73)
        at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at java.base/java.lang.Class.newInstance(Class.java:584)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:68)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:122)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:103)
        at java.base/java.lang.Thread.run(Thread.java:829)


2022-12-04 09:45:42,782 ERROR [io.qua.run.boo.StartupActionImpl] (Quarkus Main Thread) Error running Quarkus: java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:566)
        at io.quarkus.runner.bootstrap.StartupActionImpl$1.run(StartupActionImpl.java:103)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490)
        at java.base/java.lang.Class.newInstance(Class.java:584)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:68)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:42)
        at io.quarkus.runtime.Quarkus.run(Quarkus.java:122)
        at io.quarkus.runner.GeneratedMain.main(Unknown Source)
        ... 6 more
Caused by: java.lang.RuntimeException: Failed to start quarkus
        at io.quarkus.runner.ApplicationImpl.<clinit>(Unknown Source)
        ... 15 more
Caused by: org.hibernate.MappingException: Could not determine type for: org.dhbw.mosbach.ai.Bearing, at table: OArrangement, for columns: [org.hibernate.mapping.Column(bearingA)]
        at org.hibernate.mapping.SimpleValue.getType(SimpleValue.java:515)
        at org.hibernate.mapping.SimpleValue.isValid(SimpleValue.java:482)
        at org.hibernate.mapping.Property.isValid(Property.java:231)
        at org.hibernate.mapping.PersistentClass.validate(PersistentClass.java:627)
        at org.hibernate.mapping.RootClass.validate(RootClass.java:267)
        at org.hibernate.boot.internal.MetadataImpl.validate(MetadataImpl.java:359)
        at io.quarkus.hibernate.orm.runtime.recording.PrevalidatedQuarkusMetadata.validateAndWrap(PrevalidatedQuarkusMetadata.java:58)
        at io.quarkus.hibernate.orm.runtime.boot.FastBootMetadataBuilder.trimBootstrapMetadata(FastBootMetadataBuilder.java:438)
        at io.quarkus.hibernate.orm.runtime.boot.FastBootMetadataBuilder.build(FastBootMetadataBuilder.java:399)
        at io.quarkus.hibernate.orm.runtime.PersistenceUnitsHolder.createMetadata(PersistenceUnitsHolder.java:101)
        at io.quarkus.hibernate.orm.runtime.PersistenceUnitsHolder.constructMetadataAdvance(PersistenceUnitsHolder.java:73)
        at io.quarkus.hibernate.orm.runtime.PersistenceUnitsHolder.initializeJpa(PersistenceUnitsHolder.java:40)
        at io.quarkus.hibernate.orm.runtime.HibernateOrmRecorder$1.created(HibernateOrmRecorder.java:76)
        at io.quarkus.arc.runtime.ArcRecorder.initBeanContainer(ArcRecorder.java:73)
        at io.quarkus.deployment.steps.ArcProcessor$generateResources1337887788.deploy_0(Unknown Source)
        at io.quarkus.deployment.steps.ArcProcessor$generateResources1337887788.deploy(Unknown Source)
        ... 16 more



--
Tests paused
Press [r] to resume testing, [o] Toggle test output, [:] for the terminal, [h] for more options>

**Review by philippabele — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 2 above)
Comment (TheMaxens) [reply src/main/resources/application.properties:10]: Yes, I have to create a database to use Hibernate ORM with Panache.

There will be further commits.

**Comment by TheMaxens** [comment]
Before running the app with `quarkus dev`, run `docker run --rm=true --name hibernate -e POSTGRES_USER=hibernate -e POSTGRES_PASSWORD=hibernate -e POSTGRES_DB=hibernate -p 5432:5432 postgres:14.1` for starting the database.

How should the creation (POST) look like? Passing a JSON-document or just URL-parameters?

File: src/main/resources/application.properties [hunk 3]
```diff
@@ -2,4 +2,12 @@ quarkus.http.host=0.0.0.0
...
 quarkus.smallrye-openapi.info-contact-email=techsupport@example.com
 quarkus.smallrye-openapi.info-contact-name=Example API Support
-quarkus.smallrye-openapi.info-contact-url=http://exampleurl.com/contact
\ No newline at end of file
+quarkus.smallrye-openapi.info-contact-url=http://exampleurl.com/contact
+# configure your datasource
+quarkus.datasource.db-kind = postgresql
+quarkus.datasource.username = hibernate
+quarkus.datasource.password = hibernate
+quarkus.datasource.jdbc.url = jdbc:postgresql://localhost:5432/hibernate
```
Comment (philippabele) [inline src/main/resources/application.properties:10]: - Deploying DB and running quarkus app

I deployed the postgresql instance as mentioned from you.

By running `quarkus dev` I receive following error

###

__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2022-12-04 17:23:26,742 WARN  [io.agr.pool] (agroal-11) Datasource '<default>': Connection to 0.0.0.0:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.

2022-12-04 17:23:26,746 WARN  [org.hib.eng.jdb.env.int.JdbcEnvironmentInitiator] (JPA Startup Thread: <default>) HHH000342: Could not obtain connection to query metadata: org.postgresql.util.PSQLException: Connection to 0.0.0.0:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
        at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:319)
        at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49)
        at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:247)
        at org.postgresql.Driver.makeConnection(Driver.java:434)
        at org.postgresql.Driver.connect(Driver.java:291)
        at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:226)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:535)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:516)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1126)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.net.ConnectException: Connection refused (Connection refused)
        at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
        at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:412)
        at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:253)
        at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:237)
        at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
        at java.base/java.net.Socket.connect(Socket.java:609)
        at org.postgresql.core.PGStream.createSocket(PGStream.java:241)
        at org.postgresql.core.PGStream.<init>(PGStream.java:98)
        at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:109)
        at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:235)
        ... 12 more


2022-12-04 17:23:26,909 WARN  [io.agr.pool] (agroal-11) Datasource '<default>': Connection to 0.0.0.0:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
2022-12-04 17:23:26,910 WARN  [org.hib.eng.jdb.spi.SqlExceptionHelper] (JPA Startup Thread: <default>) SQL Error: 0, SQLState: 08001
2022-12-04 17:23:26,911 ERROR [org.hib.eng.jdb.spi.SqlExceptionHelper] (JPA Startup Thread: <default>) Connection to 0.0.0.0:5432 refused. Check that the hostname and port are correct and that the postmaster is accepting TCP/IP connections.
2022-12-04 17:23:26,933 ERROR [io.qua.run.Application] (Quarkus Main Thread) Failed to start application (with profile dev): java.net.ConnectException: Connection refused (Connection refused)
        at java.base/java.net.PlainSocketImpl.socketConnect(Native Method)
        at java.base/java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:412)
        at java.base/java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:253)
        at java.base/java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:237)
        at java.base/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:392)
        at java.base/java.net.Socket.connect(Socket.java:609)
        at org.postgresql.core.PGStream.createSocket(PGStream.java:241)
        at org.postgresql.core.PGStream.<init>(PGStream.java:98)
        at org.postgresql.core.v3.ConnectionFactoryImpl.tryConnect(ConnectionFactoryImpl.java:109)
        at org.postgresql.core.v3.ConnectionFactoryImpl.openConnectionImpl(ConnectionFactoryImpl.java:235)
        at org.postgresql.core.ConnectionFactory.openConnection(ConnectionFactory.java:49)
        at org.postgresql.jdbc.PgConnection.<init>(PgConnection.java:247)
        at org.postgresql.Driver.makeConnection(Driver.java:434)
        at org.postgresql.Driver.connect(Driver.java:291)
        at io.agroal.pool.ConnectionFactory.createConnection(ConnectionFactory.java:226)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:535)
        at io.agroal.pool.ConnectionPool$CreateConnectionTask.call(ConnectionPool.java:516)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at io.agroal.pool.util.PriorityScheduledExecutor.beforeExecute(PriorityScheduledExecutor.java:75)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1126)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
        at java.base/java.lang.Thread.run(Thread.java:829)


2022-12-04 17:23:26,936 INFO  [io.qua.dep.dev.IsolatedDevModeMain] (main) Attempting to start live reload endpoint to recover from previous Quarkus startup failure

--
Tests paused
Press [space] to restart, [e] to edit command line args (currently ''), [r] to resume testing, [o] Toggle test output, [:] for the terminal, [h] for more options>

###

- Testing Connection

After testing the connection, like describe [here](https://stackoverflow.com/questions/26911508/postgres-testing-database-connection-in-bash) I receive following message:

###

root@4ffa65374055:/ball-bearings-with-quarkus# pg_isready -d hibernate -h 0.0.0.0 -p 5432 -U hibernate
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = (unset),
        LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
0.0.0.0:5432 - no response
root@4ffa65374055:/ball-bearings-with-quarkus# pg_isready -d hibernate -h 127.0.0.1 -p 5432 -U hibernate
perl: warning: Setting locale failed.
perl: warning: Please check that your locale settings:
        LANGUAGE = (unset),
        LC_ALL = (unset),
        LANG = "en_US.UTF-8"
    are supported and installed on your system.
perl: warning: Falling back to the standard locale ("C").
127.0.0.1:5432 - no response

###

- next Steps

How is your development set up ? Are you developing your app inside a docker container ? I think there is an Issue Connection the quarkus App from one container to the DB container.

**Review by philippabele — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties (hunk 3 above)
Comment (philippabele) [reply src/main/resources/application.properties:10]: The Issue has to located inside the dev docker of the quarkus app.

By running `myUser@myMachine:~$ pg_isready -d hibernate -h 0.0.0.0 -p 5432 -U hibernate
0.0.0.0:5432 - accepting connections` from my host terminal. The connection works.

Comment (philippabele) [reply src/main/resources/application.properties:10]: I installed locale `apt-get -y install locales`

and set the locale

`locale-gen en_US.UTF-8 `
`update-locale LANG=en_US.UTF-8`

But the error still occurs. Now with a shorter response:

`root@4ffa65374055:/ball-bearings-with-quarkus# pg_isready -d hibernate -h 0.0.0.0 -p 5432 -U hibernate`

throws

`0.0.0.0:5432 - no response`

Comment (philippabele) [reply src/main/resources/application.properties:10]: I found the solution:

Connection from the dev container to the DB container works via the actual HostIP like

`pg_isready -d hibernate -h 192.168.178.27 -p 5432 -U hibernate`

throws

`192.168.178.27:5432 - accepting connections`

The result is, that we need to adapt the settings of the application.properties of the quarkus app and/or the linking of the container via docker network https://docs.docker.com/network/network-tutorial-standalone/

Comment (philippabele) [reply src/main/resources/application.properties:10]: Could you take care of the solution? I will review your changes as usual.

Comment (TheMaxens) [reply src/main/resources/application.properties:10]: Thank you for your troubleshooting!

I will work on the solution.

Comment (TheMaxens) [reply src/main/resources/application.properties]: I adjusted the hostname in `application.properties` via the env variable `POSTGRESQL_SERVICE_HOST`.

Copy the content of `.env.example` in your own .env file. When you are developing on your local machine the env variable has to be unset, when you are developing in a docker container the varibale has to be set to `hibernate`. You can do that via comments (#). 

The quarkus dev container and the database have to be in the same network to resolve hostnames, therefore do the following steps:
- `docker network create --driver bridge quarkus-net`
- attach already created container to the new network: `docker network connect quarkus-net <container-name>` 
 **or** create new container in the network: `docker build --build-arg GH_TOKEN=<GH-TOKEN> --build-arg username=<git-username>--build-arg email=<git-email> -t quarkus_image .` (unchanged) + `docker run --network quarkus-net --name quarkus -it quarkus_image`
- attach database container to the network: `docker run --rm=true --name hibernate --network quarkus-net -e POSTGRES_USER=hibernate -e POSTGRES_PASSWORD=hibernate -e POSTGRES_DB=hibernate -p 5432:5432 postgres:14.1`

File: .env.example [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# create your own .env file in the root of the projekt
+
+# this variable has to be set when you are developing in a docker container 
+# POSTGRESQL_SERVICE_HOST=hibernate
```
Comment (philippabele) [inline .env.example:4]: Setting this env is okay, but you have to mention the networking part as well.

If you just set this env, an quarkus dev container won't find the db with the hostname "hibernate".

Docker container are not connected by default. There are two solutions available

1. repalce "hibernate" with the host IP
2. add networking commands to the deployment of the postgres DB container

Additionally, move your comment "how to run postgres DB" into the readme or the .env.example to make this information general available

**Review by philippabele — CHANGES_REQUESTED** [review]

File: .env.example (hunk 4 above)
Comment (TheMaxens) [reply .env.example:4]: > The quarkus dev container and the database have to be in the same network to resolve hostnames, therefore do the following steps:
> 
> * `docker network create --driver bridge quarkus-net`
> * attach already created container to the new network: `docker network connect quarkus-net <container-name>`
>   **or** create new container in the network: `docker build --build-arg GH_TOKEN=<GH-TOKEN> --build-arg username=<git-username>--build-arg email=<git-email> -t quarkus_image .` (unchanged) + `docker run --network quarkus-net --name quarkus -it quarkus_image`
> * attach database container to the network: `docker run --rm=true --name hibernate --network quarkus-net -e POSTGRES_USER=hibernate -e POSTGRES_PASSWORD=hibernate -e POSTGRES_DB=hibernate -p 5432:5432 postgres:14.1`

I have already added the networking commands. Therefore I think that the containers are connected via the quarkus-net (in my dev container it worked anyway.)

Comment (philippabele) [reply .env.example:4]: Yes, you are right. The containers are connected and your set up is working.

But, where is the `docker commands` documentation ?

The comments of this issue won't be accessible for future users of your respository. They need a section in your LaTex documentation or a section in a readme.md file or somethin else to get the necessary information to get the project running.

Comment (TheMaxens) [reply .env.example:4]: Okay, now I understand.

Comment (TheMaxens) [reply .env.example:4]: I will open a new PR for the corresponding adjustments in the latex docs.

**Pull request merged**